src/Common/src/CoreLib/System/Memory.cs

```diff
@@ -73,7 +76,12 @@ public Memory(T[] array)
         public Memory(T[] array, int start, int length)
         {
             if (array == null)
-                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
+            {
+                _object = default;
+                _index = default;
+                _length = default;
+                return; // returns default
```

Comment:
It is not exactly the same case since there are bounds. Perhaps worth discussing what to do here.

If `null` is the same as empty, I think this might need to do range checks.